aws/resource_aws_ebs_snapshot_copy_test.go

```diff
@@ -93,6 +98,67 @@ func TestAccAWSEbsSnapshotCopy_withKms(t *testing.T) {
 	})
 }
 
+func TestAccAWSEbsSnapshotCopy_disappears(t *testing.T) {
+	var v ec2.Snapshot
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEbsSnapshotCopyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsEbsSnapshotCopyConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEbsSnapshotCopyExists("aws_ebs_snapshot_copy.test", &v),
+					testAccCheckEbsSnapshotCopyDisappears(&v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckEbsSnapshotCopyDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ebs_snapshot_copy" {
+			continue
+		}
+
+		resp, err := conn.DescribeSnapshots(&ec2.DescribeSnapshotsInput{
+			SnapshotIds: []*string{aws.String(rs.Primary.ID)},
+		})
+
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidSnapshot.NotFound" {
+			return nil
```

Comment:
`CheckDestroy` should not use `return nil` in the `for` loop, to ensure all resources are properly checked. 👍 

```suggestion
			continue
```

Reply:
fixed.